src/App.tsx

```diff
@@ -1,61 +1,118 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { useState } from 'react';
 
 export const App = () => {
+  const [titleValue, setTitleValue] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const todosWithUsers: Todo[] = todosFromServer.map(todo => {
+    const user = usersFromServer.find(usr => usr.id === todo.userId);
+
+    return { ...todo, user };
+  });
+
+  const [todos, setTodos] = useState(todosWithUsers);
+
+  const getNewTodoId = (todosArray: Todo[]) => {
+    const maxId = Math.max(...todosArray.map(todo => todo.id));
+
+    return maxId + 1;
+  };
+
+  const handleTitleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitleValue(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(Number(event.target.value));
+    setHasUserError(false);
+  };
+
+  const handleAddNewTodo = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    let titleHasError = false;
+    let userHasError = false;
+
+    if (!titleValue) {
+      setHasTitleError(true);
+      titleHasError = true;
+    }
+
+    if (!selectedUserId) {
+      setHasUserError(true);
+      userHasError = true;
+    }
+
+    if (titleHasError || userHasError) {
+      return;
+    }
```

Comment:
```suggestion
    if (!titleValue) {
      setHasTitleError(true);
      return;
    }

    if (!selectedUserId) {
      setHasUserError(true);
      return;
    }
```

Reply:
> 

I don't agree with you here. We don't need to use separate checkings. In the example (https://mate-academy.github.io/react_add-todo-form/), you can see that after pressing 'add' button, there're two warnings
![image](https://github.com/mate-academy/react_add-todo-form/assets/131695806/2e7e0545-8005-48b6-975b-87a5121f407f)